js/web/lib/wasm/jsep/webgpu/ops/range.ts

```diff
@@ -23,22 +23,33 @@ const createRangeProgramInfo = (start: number, limit: number, delta: number, dat
   const outputShape: number[] = [numElements];
   const outputSize = numElements;
 
-  const output = outputVariable('output', dataType, outputShape);
-  const wgslType = output.type.storage;
+  const output = outputVariable('output', dataType, outputShape.length);
+  const wgslType = output.type.value;
+  const tensorDataType = tensorDataTypeEnumToString(dataType) as ProgramUniform['type'];
+  const programUniforms: ProgramUniform[] = [
+    {type: 'uint32', data: outputSize}, {type: tensorDataType, data: start}, {type: tensorDataType, data: delta},
+    ...createTensorShapeVariables(outputShape)
+  ];
+  const uniforms: UniformsArrayType = [
+    {name: 'outputSize', type: 'u32'}, {name: 'start', type: wgslType as UniformDataElementType},
+    {name: 'delta', type: wgslType as UniformDataElementType}
+  ];
 
   const getShaderSource = (shaderHelper: ShaderHelper) => `
-        ${shaderHelper.declareVariables(output)}
+        ${shaderHelper.registerUniforms(uniforms).declareVariables(output)}
         ${shaderHelper.mainStart()}
-        ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes(outputSize)}
-        output[global_idx] = ${wgslType}(${start}) + ${wgslType}(global_idx) * ${wgslType}(${delta});
+        ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes('uniforms.outputSize')}
+        output[global_idx] = uniforms.start + ${wgslType}(global_idx) * uniforms.delta;
       }`;
   return {
     name: 'Range',
-    shaderCache: {hint: [start, limit, delta].map(x => x.toString()).join('_')},
+    shaderCache: {hint: wgslType},
```

Comment:
nit: use `dataType`. Then `output` and `wgslType` can be moved into `getShaderSource`.

Reply:
done